starter_code/src/movies.js

```diff
@@ -1,20 +1,101 @@
 /* eslint no-restricted-globals: 'off' */
 // Turn duration of the movies from hours to minutes 
 
-
 // Get the average of all rates with 2 decimals 
-
+function ratesAverage(movies){
+  var totalRates = 
+  movies.reduce(function(acc,elem){
+    return acc += parseFloat(elem.rate);
+  },0)
+  return Math.round(totalRates/(movies.length)*100)/100
+}
+ratesAverage(movies)
 
 // Get the average of Drama Movies
 
+function dramaMoviesRate(movies){
+    var dramaMovie = 
+    movies.filter(function(oneMovie){
+    return ((oneMovie.genre.indexOf("Drama")) > -1);
+  })
+    return ratesAverage(dramaMovie);
+  }
+  dramaMoviesRate(movies)
```

Comment:
L'erreur Jasmine "Return Average even if one of the movies does not have rate!" attend qu'on considère que le rate est 0 si une movie n'a pas de rate (ça n'a pas de sens, je suis d'accord!) donc tu peux ajoter ça dans le reduce de ta fonction rates average au dessus pour t'en débarrasser :) : 
```js
  if (elem.rate === "") {
      elem.rate = 0;
    }
```

Pour la dernière erreur, il suffit de rajouter un if statement pour gérer le cas où il n'y a pas de drama movie : 
```js
function dramaMoviesRate(movies) {
  var dramaMovie = movies.filter(function(oneMovie) {
    return oneMovie.genre.indexOf("Drama") > -1;
  });

  if (dramaMovie.length === 0) {
    return undefined;
  }
  return ratesAverage(dramaMovie);
}
```